staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth_test.go

```diff
@@ -63,6 +63,12 @@ func TestBasicAuth(t *testing.T) {
 			ExpectedPassword: "",
 			ExpectedErr:      true,
 		},
+		"empty username": {
+			ExpectedCalled:   true,
+			ExpectedUsername: "",
+			ExpectedPassword: "mypassword",
```

Comment:
any testcase with non-nil `ExpectedUsername`/`ExpectedPassword` will trigger an [errInvalidAuth](https://github.com/kubernetes/kubernetes/blob/1719a9efc393dd48f1414d4949fb72000266a76e/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth.go#L49), as long as [Password](https://github.com/kubernetes/kubernetes/blob/1719a9efc393dd48f1414d4949fb72000266a76e/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth_test.go#L49) is unassigned

this file is for testing if different [Password interfaces](https://github.com/kubernetes/kubernetes/blob/1719a9efc393dd48f1414d4949fb72000266a76e/staging/src/k8s.io/apiserver/pkg/authentication/authenticator/interfaces.go#L42)' responses get handled by [Authenticator](https://github.com/kubernetes/kubernetes/blob/1719a9efc393dd48f1414d4949fb72000266a76e/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/request/basicauth/basicauth.go#L39) properly. adding a case that triggers same response from an existing `Password` doesn't add much value IMO

Reply:
Thank for the review, but what if there is a case when username is empty and password will be anything let's say "mypass:withcolon" ? Couldn't we should handle this case too ?